### PR TITLE
Initial work toward exposing the Java native interface APIs in Dart

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: sky/engine/bindings/dart_ui.cc [hunk 1]
```diff
@@ -50,6 +51,7 @@ void DartUI::InitForGlobal() {
     CanvasImage::RegisterNatives(g_natives);
     CanvasPath::RegisterNatives(g_natives);
     ColorFilter::RegisterNatives(g_natives);
+    DartJni::RegisterNatives(g_natives);
```
Comment (abarth) [inline sky/engine/bindings/dart_ui.cc]: #ifdef OS(ANDROID) ?

Also, please add a TODO about moving this into `dart:jni`.  It's ok to have it in `dart:ui` for the time being.

File: sky/engine/core/dart/jni.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart_ui;
```
Comment (abarth) [inline sky/engine/core/dart/jni.dart]: Can you put this file in `sky/engine/bindings/jni` ?  It should eventually becomes its own built-in library.

File: sky/shell/platform/android/library_loader.cc [hunk 3]
```diff
@@ -35,15 +35,22 @@ bool RegisterJNI(JNIEnv* env) {
                                arraysize(kSkyRegisteredMethods));
 }
 
+bool InitJNI() {
+  return sky::shell::InitJniSkyMain();
```
Comment (abarth) [inline sky/shell/platform/android/library_loader.cc]: Why bounce through sky/shell?  Seems like you could go to blink::DartJni::InitJni directly.

**Comment by abarth** [comment]
LGTM

File: sky/engine/bindings/dart_ui.cc (hunk 1 above)
Comment (jason-simmons) [inline sky/engine/bindings/dart_ui.cc]: done

File: sky/engine/core/dart/jni.dart (hunk 2 above)
Comment (jason-simmons) [inline sky/engine/core/dart/jni.dart]: done

File: sky/shell/platform/android/library_loader.cc (hunk 3 above)
Comment (jason-simmons) [inline sky/shell/platform/android/library_loader.cc]: done

**Comment by abarth** [comment]
LGTM

**Pull request merged**